mewarpx/mewarpx/diags_store/particle_histogram_diagnostic.py

```diff
@@ -0,0 +1,181 @@
+import logging
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+from pywarpx import callbacks
+
+from mewarpx import assemblies
+from mewarpx.diags_store import diag_base
+from mewarpx.mwxrun import mwxrun
+from mewarpx.utils_store import parallel_util
+
+logger = logging.getLogger(__name__)
+
+class ParticleHistDiag(diag_base.WarpXDiagnostic):
```

Comment:
Overall I don't like the loss of generality that comes here by assuming this is storing 3 position dimensions and nothing else, and that boundaries are simulation boundaries, especially since the more generic code is available:

* For say a small embedded boundary, you would want limits of bins to be the limits of that EB extent, not the whole simulation, which will create lots of useless bins outside the EB.
* There's lots of histogram options that are eliminated here that wouldn't be hard to add - velocity vs position relations, or positions at birth which can be saved to a new pid and are nice for ionization statistics.
* Handling different assemblies through if statements is not very scalable/readable/modular

I would make ParticleHistDiag a more generic histogramming tool, and then create a child class with specific values for which dimensions etc. are kept if you do want this specificity easily available.